packages/next/server/web/spec-extension/response.ts

```diff
@@ -13,8 +20,29 @@ export class NextResponse extends Response {
     url?: NextURL
   }
 
-  constructor(body?: BodyInit | null, init: ResponseInit = {}) {
-    super(body, init)
+  constructor(
+    body?: BodyInit | { [k: string]: any } | null,
+    init: ResponseInit = {}
+  ) {
+    if (
+      body &&
+      typeof body === 'object' &&
+      !isArrayBuffer(body) &&
+      !isArrayBufferView(body) &&
+      !isBlob(body) &&
+      !isFormData(body) &&
+      !isReadableStream(body)
+    ) {
+      super(JSON.stringify(body), {
```

Comment:
Can we update the doc [here](https://nextjs.org/docs/api-reference/next/server#nextresponse) mentioning this is specific to `NextResponse` and isn't specific to the [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response) API. 

Also just an idea but would it make sense to have this be a separate static method like `NextResponse.json()` instead so that we don't have to keep updating this check with the default handled types?

Reply:
I kind of like that idea better 🤔 It makes sense considering `.rewrite` and `.redirect` are doing a very similar thing. WDYT @Kikobeats ?

Reply:
I like the idea 👍